tools/clang/unittests/HLSL/CompilerTest.cpp

```diff
@@ -3218,6 +3220,77 @@ TEST_F(CompilerTest, PreprocessWhenExpandTokenPastingOperandThenAccept) {
                        text.c_str());
 }
 
+TEST_F(CompilerTest, PreprocessWithDebugOptsThenOk) {
+  // Make sure debug options, such as -Zi and -Fd,
```

Comment:
Can't tell if you have a test for `-P -Zi -Qembed_debug`, but would be good to have that since it seemed to cause the crash.

Reply:
The new interface is invoked with `-P` by `DxcCompilerAdapter::WrapCompile()`, and the original repro was in fact using this older interface.  `-Qembed_debug` will have no effect on the repro.

Reply:
In any case, I added `-Qembed_debug` to the test, since I was modifying the `writePDB` condition to simplify it when I noticed the `produceFullContainer` flag already contained what was needed for the mode options.